ceno_emul/src/platform.rs

```diff
@@ -60,21 +69,21 @@ impl Platform {
     // Startup.
 
     pub const fn pc_base(&self) -> Addr {
-        self.rom.start
+        self.prog_code.start
     }
 
     // Permissions.
 
     pub fn can_read(&self, addr: Addr) -> bool {
-        self.is_rom(addr) || self.is_ram(addr) || self.is_pub_io(addr) || self.is_hints(addr)
+        self.can_write(addr)
     }
 
     pub fn can_write(&self, addr: Addr) -> bool {
-        self.is_ram(addr)
+        self.is_ram(addr) || self.is_pub_io(addr) || self.is_hints(addr)
```

Comment:
`hints` (and pub io) are supposed be read-only.

Reply:
They are not read-only so this matches correctly.

Reply:
OK.  I created an issue at https://github.com/scroll-tech/ceno/issues/751

Btw, we do have read only sections in the ELF already, and we should honour them.